genesis/process/shardGenesisBlockCreator.go

```diff
@@ -445,25 +445,38 @@ func deployInitialSmartContract(
 		return err
 	}
 
-	argDeploy := intermediate.ArgDeployProcessor{
-		Executor:       txExecutor,
-		PubkeyConv:     arg.PubkeyConv,
-		BlockchainHook: processors.blockchainHook,
-		QueryService:   processors.queryService,
-	}
-	deployProc, err := intermediate.NewDeployProcessor(argDeploy)
-	if err != nil {
-		return err
-	}
-
 	switch sc.GetType() {
 	case genesis.DelegationType:
 		deployMetrics.numDelegation++
-	default:
+		argDeploy := intermediate.ArgDeployProcessor{
+			Executor:       txExecutor,
+			PubkeyConv:     arg.PubkeyConv,
+			BlockchainHook: processors.blockchainHook,
+			QueryService:   processors.queryService,
+		}
+		deployProc, err := intermediate.NewDeployProcessor(argDeploy)
+		if err != nil {
+			return err
+		}
+
+		return deployProc.Deploy(sc)
+	case genesis.DNSType:
 		deployMetrics.numOtherTypes++
+		argDeployLibrary := intermediate.ArgDeployLibrarySC{
+			Executor:         txExecutor,
+			PubkeyConv:       arg.PubkeyConv,
+			BlockchainHook:   processors.blockchainHook,
+			ShardCoordinator: arg.ShardCoordinator,
+		}
+		deployLibrary, err := intermediate.NewDeployLibrarySC(argDeployLibrary)
+		if err != nil {
+			return err
+		}
+
+		return deployLibrary.Deploy(sc)
```

Comment:
We need the default back here in order to deploy other smart contracts at genesis time (that might not need a special configuration)

Reply:
I do not see what other special cases we might have - if they are special most probably they will have their own type.